services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3809,27 +3799,20 @@ private void setRmdChunkingEnabled(String clusterName, String storeName, boolean
   void setIncrementalPushEnabled(String clusterName, String storeName, boolean incrementalPushEnabled) {
     storeMetadataUpdate(clusterName, storeName, store -> {
       VeniceControllerClusterConfig config = getHelixVeniceClusterResources(clusterName).getConfig();
-      if (incrementalPushEnabled) {
+      if (incrementalPushEnabled || store.isHybrid()) {
         // Enabling incremental push
+        store.setNativeReplicationEnabled(config.isNativeReplicationEnabledAsDefaultForHybrid());
```

Comment:
I am wondering if all these existing configs are still on reverse order... like we should go from NR enabled -> inc push enabled not the other way... but I guess it is fine for now as NR is enabled everywhere.

Reply:
It is by design...  we determine whether to enable NR based on the store type and not vice versa.

Reply:
It is by design...  we determine whether to enable NR based on the store type and not vice versa.